### PR TITLE
Improve usability of configuration for ejected images

### PR DESCRIPTION
**Pull request by ekerfelt**
This PR introduces an option to define an infrastructure config file when eject an image using a `--config` flag, e.g.

`encoredev eject docker --config infra.config.json test`

The infra config will be embedded in the exported docker image and replaces the need for the ENCORE_RUNTIME_CONFIG and ENCORE_SECRETS env variable

### CODE REVIEW

File: cli/cmd/encore/eject.go [hunk 1]
```diff
@@ -49,18 +50,26 @@ func init() {
 	dockerEjectCmd.Flags().StringVar(&p.Goos, "os", p.Goos, "target operating system. One of (linux, darwin, windows)")
 	dockerEjectCmd.Flags().StringVar(&p.Goarch, "arch", p.Goarch, "target architecture. One of (amd64, arm64)")
 	dockerEjectCmd.Flags().BoolVar(&p.CgoEnabled, "cgo", false, "enable cgo")
+	dockerEjectCmd.Flags().BoolVar(&p.SkipInfraConf, "skip-infra-config", false, "do not read or generate a infra configuration file")
```
Comment (eandre) [inline cli/cmd/encore/eject.go:53]: If the argument is `--config`, should this be `--skip-config` instead?

File: runtimes/go/appruntime/exported/config/parse.go [hunk 2]
```diff
@@ -52,61 +52,362 @@ func ParseRuntime(config, processCfg, deployID string) *Runtime {
...
+	if val == zero {
+		return def
+	}
+	return val
+}
+
+// ParseRuntime parses the Encore runtime config.
+func ParseRuntime(runtimeConfig, processCfg, infraCfgPath, deployID string) *Runtime {
+	var cfg *Runtime
+	if infraCfgPath == "" {
```
Comment (eandre) [inline runtimes/go/appruntime/exported/config/parse.go:395]: Shouldn't this be:

```suggestion
	if infraCfgPath != "" {
```

File: cli/cmd/encore/eject.go (hunk 1 above)
Comment (ekerfelt) [reply cli/cmd/encore/eject.go:53]: yes!

File: runtimes/go/appruntime/exported/config/parse.go (hunk 2 above)
Comment (ekerfelt) [reply runtimes/go/appruntime/exported/config/parse.go:395]: yes

File: runtimes/core/src/lib.rs [hunk 3]
```diff
@@ -440,6 +451,18 @@ impl Display for ParseError {
 
 impl std::error::Error for ParseError {}
 
+fn infra_config_from_env() -> Result<Option<runtimepb::RuntimeConfig>, ParseError> {
+    let cfg_path = match std::env::var("ENCORE_INFRA_CONFIG_PATH") {
+        Ok(cfg) => cfg,
+        Err(_) => return Ok(None),
```
Comment (eandre) [inline runtimes/core/src/lib.rs:457]: How do you feel about:
```suggestion
        Err(std::env::VarError::NotPresent) => return Ok(None),
        Err(e) => return Err(ParseError::EnvVar(e)),
```

File: docs/how-to/self-host.md [hunk 4]
```diff
@@ -32,72 +35,80 @@ This includes configuring things like:
...
-  "env_type": "development",
-  "env_cloud": "local",
-  "deploy_time": "2024-04-09T16:25:06.502476123Z",
-  "cors": {
-    "allow_origins_without_credentials": ["*"]
-  },
-  "service_auth": [{ "method": "noop" }],
-  "gateways": [
+  "$schema": "https://encore.dev/schemas/infra.schema.json",
+  "app_id": "my-app",
```
Comment (eandre) [inline docs/how-to/self-host.md]: This is missing the metadata key, right?

**Review by eandre — APPROVED** [review]

**Pull request merged**